### PR TITLE
Adds go-kit metrics middleware

### PR DESCRIPTION
**Pull request by rjfonseca**
This commits introduces the Metrifier and a go-kit metrics middleware
that uses the Metrifier.

The Metrifier is a struct that helps calculating prometheus metrics for
an arbitraty piece of code.

The go-kit metrics endpoint uses de Metrifier to calculate latency and
count for endpoints. It allows for adding extra matrics labels by using
decoders.

The logging middleware was changed to be initialized the same way as the
metrics middleware for consistency.

### CODE REVIEW

File: gokitmiddlewares/loggingmiddleware/config.go [hunk 1]
```diff
@@ -13,6 +13,10 @@ import (
 // loggingmiddleware. All fields should be set unless marked
 // as optional.
 type Config struct {
+	// Name is the name of the endpoint.
+	// It must not be empty abnd will be logged as endpoint_name.
```
Comment (eric-reis) [inline gokitmiddlewares/loggingmiddleware/config.go:17]: Nit: `abnd`

File: gokitmiddlewares/loggingmiddleware/middleware.go [hunk 2]
```diff
@@ -6,23 +6,20 @@ import (
...
-	if err != nil {
-		panic(err)
-	}
-	return m
+func MustNew(c Config) endpoint.Middleware {
+	return gokitmiddlewares.Must(New(c))
 }
 
 // New returns a new go-kit logging middleware with the given name and configuration.
 // It will panic if the name is empty.
```
Comment (eric-reis) [inline gokitmiddlewares/loggingmiddleware/middleware.go:20]: Nit: Actually, `New` will return an error instead of panicking.

File: gokitmiddlewares/must.go [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package gokitmiddlewares
+
+import "github.com/go-kit/kit/endpoint"
+
+func Must(e endpoint.Middleware, err error) endpoint.Middleware {
```
Comment (eric-reis) [inline gokitmiddlewares/must.go:5]: Don't know why the linter didn't point this up. Missing godoc comment.

File: metrifier/config.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	//
+	// ConstLabels are only used rarely. In particular, do not use them to
+	// attach the same labels to all your metrics. Those use cases are
+	// better covered by target labels set by the scraping Prometheus
+	// server, or by one specific metric (e.g. a build_info or a
+	// machine_role metric). See also
+	// https://prometheus.io/docs/instrumenting/writing_exporters/#target-labels,-not-static-scraped-labels
+	ConstLabels stdprometheus.Labels
+
+	// The bellow fields applies only to summary
```
Comment (eric-reis) [inline metrifier/config.go:52]: Nit: by grouping in a inner struct, the grouping might become clearer than with a label comment

File: metrifier/metrifier.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+	kitprometheus "github.com/go-kit/kit/metrics/prometheus"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	labelErrorCode = "error_code"
+)
+
+// Metrifier is a struct that helps metrify any computation using
+// two Promethus metrics: execution_count and execution_latency_seconds.
```
Comment (eric-reis) [inline metrifier/metrifier.go:16]: Nit: this documentation can be improved by saying that there's one metrifer per subject being measured. I.e. two distinct output queues will each have its own metrifier.

File: trace/id.go [hunk 6]
```diff
@@ -17,6 +17,7 @@ type Generator interface {
 
 var traceIDGen Generator
 
+// nolint: gosec
```
Comment (andremissaglia) [inline trace/id.go:20]: Why is this here? could you comment on what was the issue with the linter, and why it can be ignored?

Maybe you can put this directly over the line that caused the problem? I feel that gosec is too important to ignore in the whole function.

File: gokitmiddlewares/metricsmiddleware/middleware.go [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+		return nil, err
+	}
+
+	return func(next endpoint.Endpoint) endpoint.Endpoint {
+		return func(ctx context.Context, req interface{}) (resp interface{}, err error) {
+			s := m.Begin()
+			defer func() {
+				s.WithLabels(makeLabels(ctx, c, req, resp, err)).End(err)
+			}()
+			return next(ctx, req)
```
Comment (andremissaglia) [inline gokitmiddlewares/metricsmiddleware/middleware.go:39]: I liked it. Nice and clean.

File: trace/id.go (hunk 6 above)
Comment (rjfonseca) [reply trace/id.go:20]: gosec complains about math.Rand instead of crypto.Rand, but here it's ok, because math.Rand is faster and we are not using Rand for security purpose.. I will add this explanation as a comment also.

File: gokitmiddlewares/metricsmiddleware/middleware.go [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+	"errors"
+
+	"github.com/arquivei/foundationkit/gokitmiddlewares"
+	"github.com/arquivei/foundationkit/metrifier"
+	"github.com/go-kit/kit/endpoint"
+)
+
+// LabelDecoder defines a functions that generates a label value by processing the
+// endpoint's request and response.
+type LabelDecoder func(ctx context.Context, req, resp interface{}, err error) string
```
[the remark this answers is not in the document]
Comment (rjfonseca) [reply gokitmiddlewares/metricsmiddleware/middleware.go:14]: I don't think the slowness would be an issue here, but having a single function could be more efficient.

The idea of having separate decoders is to allow for reusing decoders and also easing the configuration of the middleware. If the decoder is a single function, the developer must fill the Metrifier.ExtraLabels manually and the decoder must return the same keys or the code will panic (I really should put a recover in the middleware). The current code will fill the ExtraLabels for the developer.

Heres a real example that extracts a single label from the endpoitn request:
`		metricsConfig.LabelDecoders = map[string]metricsmiddleware.LabelDecoder{
			"namespace": func(_ context.Context, req, _ interface{}, _ error) string {
				r := req.(request.Request)
				if r.Namespace == nil || *r.Namespace == "" {
					return "UNKNOWN"
				}
				return *r.Namespace
			},
		}`
